File: flight-board-ui/src/types/flight.ts
export type FlightStatus =
  | 'Scheduled'
  | 'Boarding'
  | 'Departed'
  | 'Landed';

export interface Flight {
  id: string;
  flightNumber: string;
  destination: string;
  departureTime: string; // ISO string
  gate: string;
  status: FlightStatus;
}

export interface FlightFormValues {
  flightNumber: string;
  destination: string;
  departureTime: string;
  gate: string;
} 

File: FlightBoardAPI/FlightBoardAPI/Domain/Flight.cs
namespace FlightBoard.Domain;

public class Flight
{
    public int Id { get; set; }
    public string FlightNumber { get; set; } = null!;
    public string Destination { get; set; } = null!;
    public DateTime DepartureTime { get; set; }
    public string Gate { get; set; } = null!;

}
